scripts/test_mpc

```diff
@@ -4,42 +4,112 @@ set -x
-phase2_file=_test_phase2-${QAP_DEGREE}.bin
+
+transcript_file=_test_transcript.bin
+challenge_0_file=_test_challenge_0.bin
+response_1_file=_test_response_1.bin
+response_digest_1_file=_test_response_digest_1.bin
+challenge_1_file=_test_challenge_1.bin
+response_2_file=_test_response_2.bin
+response_digest_2_file=_test_response_digest_2.bin
+challenge_2_file=_test_challenge_2.bin
+response_3_file=_test_response_3.bin
+response_digest_3_file=_test_response_digest_3.bin
+challenge_3_file=_test_challenge_3.bin
+final_phase2_file=${challenge_3_file}
+invalid_response_digest_file=_test_response_digest_invalid.bin
+
 keypair_file=_test_keypair-${QAP_DEGREE}.bin
 
+rm -f ${transcript_file}
+
 # Dummy pot data
 ${POT} --dummy ${pot_file} ${QAP_DEGREE}
 
 # Compute lagrange points
 ${POT} --out ${lagrange_file} ${pot_file} ${QAP_DEGREE}
 
 # Generate the linear combination
-${MPC} \
-       linear-combination --out ${linear_combination_file} \
+${MPC} linear-combination --out ${linear_combination_file} \
        ${pot_file} ${lagrange_file}
 
-# Create a dummy phase2 file
-${MPC} \
-       dummy-phase2 --out ${phase2_file} ${linear_combination_file}
+# Begin Phase2 MPC and run some rounds, accumulating transcript.
+${MPC} phase2-begin --out ${challenge_0_file} ${linear_combination_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_1_file} \
+       --out ${response_1_file} ${challenge_0_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_1_file} \
+       ${challenge_0_file} ${response_1_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_2_file} \
+       --out ${response_2_file} ${challenge_1_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_2_file} \
+       ${challenge_1_file} ${response_2_file}
+
+${MPC} phase2-contribute \
+       --skip-user-input \
+       --digest ${response_digest_3_file} \
+       --out ${response_3_file} ${challenge_2_file}
+${MPC} phase2-verify-contribution \
+       --transcript ${transcript_file} \
+       --new-challenge ${challenge_3_file} \
+       ${challenge_2_file} ${response_3_file}
+
+# Verify the transcript
+${MPC} phase2-verify-transcript \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+
+# Verify the transcript, checking for each contribution
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_1_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_2_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+${MPC} phase2-verify-transcript \
+       --digest ${response_digest_3_file} \
+       ${challenge_0_file} ${transcript_file} ${challenge_3_file}
+
+# Invalid contribution check
+echo 00000000 >> ${invalid_response_digest_file}
```

Comment:
Why writing `00000000` in the file?

Reply:
This, along with the next line, creates a digest file that is 4 zero bytes followed by the correct contribution hash (which, with high probability, will be read as a digest which is not in the transcript).  This checks that the `phase2-verify-transcript` command fails when used to check for a contribution that is not in the transcript.